src/FsAutoComplete/CodeFixes/ImplementInterface.fs

```diff
@@ -0,0 +1,375 @@
+      //
+      // interface IDisposable with
+      //   (*foo bar*)member 
+      //     _.Dispose () = ()
+      // ```
+      // -> first non-comment is indentation required by F#
+      // But valid:
+      // ```fsharp
+      // interface IDisposable with
+      //     (*foo bar*)member 
+      //         (*baaaaz*)_.Dispose () = ()
+      //
+      // interface IDisposable with
+      //     (*foo bar*)member 
+      //   (*baaaaaaaaaz*)_.Dispose () = ()
+      // ```
+      // (`_` (this) behind `member`)
+
+      let startCol =
+        // attribute must be first
+        attributes
+        |> List.tryHead
+        |> Option.map (fun attr -> attr.Range.StartColumn)
+        |> Option.orElseWith (fun _ ->
+            // leftmost `member` or `override` (and just to be sure: `default`, `abstract` or `static`)
+            match memberFlags with
+            | Some memberFlags ->
+                let trivia = memberFlags.Trivia
+                [
+                  trivia.StaticRange
+                  trivia.MemberRange
+                  trivia.OverrideRange
+                  trivia.AbstractRange
+                  trivia.DefaultRange
+                ]
+                |> List.choose id
+                |> List.map (fun r -> r.StartColumn)
+                   // List.tryMin
+                |> List.fold (fun m c ->
+                    match m with
+                    | None -> Some c
+                    | Some m -> 
+                        min c m
+                        |> Some
+                ) None
+            | None -> None
+        )
+        |> Option.defaultValue
+            // fallback: start of head pat (should not happen -> always `member`)
+            headPat.Range.StartColumn
+
+      let insertPos = expr.Range.End
+
+      { 
+        StartColumn = startCol 
+        InsertAt = insertPos
+        InsertWith = false
+      }
+      |> Some
+  | None ->
+      // align with `interface` or `new`
+      //    no attributes on interface allowed
+      // insert after `with` or identifier
+      match tryFindInterfaceStartAndWith interfaceData.Range.End ast with
+      | None -> None
+      | Some (startPos, withRange) ->
+          let startCol = startPos.Column + indentationSize
+          let insertPos =
+            withRange
+            |> Option.map (fun r -> r.End)
+            |> Option.defaultValue interfaceData.Range.End
+
+          { 
+            StartColumn = startCol 
+            InsertAt = insertPos
+            InsertWith = withRange |> Option.isNone
+          }
+          |> Some
+
+let titleWithTypeAnnotation = "Implement interface"
+let titleWithoutTypeAnnotation = "Implement interface without type annotation"
+
+/// codefix that generates members for an interface implementation
+let fix
+  (getParseResultsForFile: GetParseResultsForFile)
+  (getProjectOptionsForFile: GetProjectOptionsForFile)
+  (indentationSize: int, objectIdentifier: string, methodBody: string)
```

Comment:
I'd suggest wrapping the configuration for the fix in a type, and passing in a function that can get an instance of that type. The current way prevents changes to configuration from being updated over the lifetime of the FSAC process. Passing in a `unit -> ImplementInterfaceConfig` instead allows for that to change.

Reply:
done